meson.build

```diff
@@ -0,0 +1,177 @@
+# This file is part of libmicrodns.
+#
+# Copyright © 2019 VideoLabs SAS
+#
+# Author: Mathieu Duponchelle <mathieu@centricular.com>
+#
+#########################################################################
+# libmicrodns is released under LGPLv2.1 (or later) and is also available
+# under a commercial license.
+#########################################################################
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
+
+project('microDNS', ['c'],
+  version : '0.1.0',
+  meson_version : '>= 0.51.0',
+  default_options : ['warning_level=2',
+                     'buildtype=release',
+                     'b_ndebug=if-release'])
+
+mdns_version = meson.project_version()
+
+ver_arr = mdns_version.split('.')
+
+mdns_major_version = ver_arr[0]
+mdns_minor_version = ver_arr[1]
+mdns_micro_version = ver_arr[2]
+
+cc = meson.get_compiler('c')
+
+warning_flags = [
+    '-Wall',
+    '-Wsign-compare',
+    '-Wextra',
+    '-Wstrict-aliasing',
+    '-Wstrict-overflow',
+    '-Wformat=2',
+    '-Wno-unused-parameter',
+    '-pipe',
+    '-Wcast-align',
+    '-Wpointer-arith',
+    '-Wmissing-prototypes',
+    '-Wwrite-strings',
+    '-Wlogical-op',
+]
+
+add_project_arguments(cc.get_supported_arguments(warning_flags), language: 'c')
+
+cdata = configuration_data()
+
+deps = []
+
+host_system = host_machine.system()
+
+if host_system == 'windows'
+    deps += [cc.find_library('ws2_32')]
+    deps += [cc.find_library('iphlpapi')]
+endif
+
+inet_ntop_src = '''
+#ifdef _WIN32
+#include <ws2tcpip.h>
+#include <windows.h>
+# if _WIN32_WINNT < 0x600
+#  error Needs vista+
+# endif
+#else
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#endif
+int main() {}
+'''
+
+cdata.set10('HAVE_INET_NTOP', cc.links(inet_ntop_src, dependencies: deps))
+
+poll_src = '''
+#include <stddef.h>
+#ifdef _WIN32
+#include <winsock2.h>
+#include <windows.h>
+# if _WIN32_WINNT < 0x600
+#  error Needs vista+
+# endif
+#else
+#include <poll.h>
+#endif
+int main() {}
+'''
+
+cdata.set10('HAVE_INET_NTOP', cc.links(inet_ntop_src, dependencies: deps))
+cdata.set10('HAVE_POLL', cc.links(poll_src, dependencies: deps))
+
+configure_file(output : 'config.h', configuration : cdata)
+
+c_args = ['-DHAVE_CONFIG_H']
```

Comment:
Nothing you should address in this PR IMO, but do we still need this? Is there any usecase where it makes sense to not have a config.h and not error out in that case?

Reply:
That's a pretty traditional construct fwiw, haven't ever really paid it much thought but <https://en.wikipedia.org/wiki/Prefix_header> has info on the pattern

Reply:
please resolve if you're OK with that :)

Reply:
Not sure how to resolve this on Github… there is no button to mark this as resolved…